### PR TITLE
Allow using an existing Validator instance

### PR DESCRIPTION
**Pull request by ruromero**
Signed-off-by: Ruben Romero Montes <rromerom@redhat.com>

* Use jakarta.el instead of javax.el 
* Allow providing an existing Validator instance
* Refactor some fuction calls

Follow up PR for #65

### CODE REVIEW

**Comment by ruromero** [comment]
@fabiojose can you review my changes? I hope you don't mind the small cleanup.
Thanks in advance

**Comment by fabiojose** [comment]
Wow! it's big!

File: api/src/main/java/io/cloudevents/format/BinaryMarshaller.java [hunk 1]
```diff
@@ -53,7 +53,7 @@ private BinaryMarshaller() {}
 		return new Builder<A, T, P, H>();
 	}
 	
-	public static interface AttributeMarshalStep<A extends Attributes, T, P, H> {
+	public interface AttributeMarshalStep<A extends Attributes, T, P, H> {
```
Comment (fabiojose) [inline api/src/main/java/io/cloudevents/format/BinaryMarshaller.java]: Is it better no static?

File: api/src/main/java/io/cloudevents/format/BinaryUnmarshaller.java [hunk 2]
```diff
@@ -219,10 +220,8 @@ private BinaryUnmarshaller() {}
 			A attributes = attributeUnmarshaller.unmarshal(attributesMap);
 			
 			T data = attributes.getMediaType()
-				.map((mime) -> {
-					return dataUnmarshallers.get(mime);
-				})
-				.filter((un) -> null != un)
+				.map((mime) -> dataUnmarshallers.get(mime))
```
Comment (fabiojose) [inline api/src/main/java/io/cloudevents/format/BinaryUnmarshaller.java]: nice!

File: api/src/main/java/io/cloudevents/v02/AttributesImpl.java [hunk 3]
```diff
@@ -170,7 +170,7 @@ public static AttributesImpl unmarshal(Map<String, String> attributes) {
 		
 		URI schemaurl = 
 			Optional.ofNullable(attributes.get(ContextAttributes.schemaurl.name()))
-			.map(schema -> URI.create(schema))
+			.map(URI::create)
```
Comment (fabiojose) [inline api/src/main/java/io/cloudevents/v02/AttributesImpl.java]: Thank you! less verbose

**Comment by fabiojose** [comment]
That was a nice cleanup! Congrats!

In general: LGTM!

File: api/src/main/java/io/cloudevents/format/BinaryMarshaller.java (hunk 1 above)
Comment (ruromero) [reply api/src/main/java/io/cloudevents/format/BinaryMarshaller.java]: Nested interfaces are static by default so this modifier was redundant

**Comment by ruromero** [comment]
@fabiojose can we get it merged or are you waiting for another person to review it as well?
Thanks!

**Pull request merged**